### ISSUE TITLE
Errors when attempting to edit a data-set with ~16k members

### ISSUE THREAD

**Issue by tucker01** [zowe/vscode-extension-for-zowe#111]
I have a data-set with ~16k members. When attempting to open in the files tree, I get the following error popup:

```
Retrieving response from zowe.List Error: The get request appeared to succeed, but the response was not in the expected format: Unexpected token  in JSON at position 2724667
```

**Comment by Colin-Stone** [comment]
From the call yesterday it sounded as if this is an issue with the API's available and would need to be addressed by managing the download. @kristinochka Is that correct?.

**Comment by kristinochka** [comment]
@tucker01 Are you trying to expand a dataset with 16 thousand members and they all fail to load? Can you provide a screenshot or gif?

**Comment by kristinochka** [comment]
Colin, yes, I think the issue is the API is trying to load them all at once and fails.

**Comment by Colin-Stone** [comment]
Maybe this is an issue that we should raise UX team as it's a problem that requires an API design/enhancement that would benefit all users @jordanCain any thoughts?

**Comment by jordanCain** [comment]
@Colin-Stone This is definitely poor UX, I know the zosmf endpoint allows a limit to the number of members to be fetched, it defaults to 1000 so would be interesting to know if that's overwritten or this bug would be present for any dataset with a large number of members not just in the region of 16k.

I don't see anyway to do paging with the get members endpoint in zosmf so even if we restricted the get to 100 members at a time I don't think there's a way for us to say "and now the next 100 etc. " so engineering a new API would be required if we wanted to pursue paging.

A quick path to a better UX could be to direct users to also use a member search parameter which zosmf supports.

**Comment by jelaplan** [comment]
Requiring a filter value to reduce the count seems like it would help the user to get results back. 

The error message doesn't convey the problem. Could we change that message or is that something zOSMF delivers and so out of our hands?

**Comment by tucker01** [comment]
We did some more investigation and in this case, it seems like the error is actually caused by a member with a binary name in the data set. z/OSMF is returning what looks like garbage for that member, causing the Zowe CLI APIs to throw an exception on the JSON parsing. 

I tried some other data-sets with ~2k members and it seemed to retrieve them just fine.

**Comment by tucker01** [comment]
Snippet from the response:
```
  {"member":"Ú¾H;¶®"}
],"returnedRows":16367,"JSONversion":1}
```

**Comment by Colin-Stone** [comment]
@tucker01 Is this still an issue that needs addressing?. I think the filtering makes sense but is there another problem to fix with the binary data?.

**Comment by tucker01** [comment]
The length of the list here was a "red-herring". The real error was caused by a binary member name. I'm not sure how we can address this... but we probably have to in the CLI as well.

**Comment by MikeBauerCA** [comment]
At a minimum, this should be documented as a known issue prior to 1.0 MVP

**Comment by MikeBauerCA** [comment]
@tucker01 can you summarize the remaining issue here. @BrandonJenkins14 could you document this as a known issue. We should then close this issue and open up a new one as the title is misleading.
